pkg/sharing/secret_exports.go

```diff
@@ -171,11 +180,58 @@ func (es exportedSecret) Matches(matcher SecretMatcher, nsIsExcludedFromWildcard
 			return false
 		}
 	}
+
 	if len(matcher.FromNamespace) > 0 {
 		if matcher.FromNamespace != es.secret.Namespace {
 			return false
 		}
 	}
+
+	selectors := es.export.Spec.ToSelectorMatchFields
+	if len(selectors) > 0 {
+		nsName := matcher.ToNamespace
+		query := types.NamespacedName{
+			Name: nsName,
+		}
+		namespace := corev1.Namespace{}
+		err := matcher.SecretImportReconciler.client.Get(matcher.Ctx, query, &namespace)
```

Comment:
I would prefer to see a client specifically for getting this namespace (which can be expanded in the future for other usecases) rather than re-using the import reconciler's client. Definitely feels like a borrowing of something we don't really need.

Reply:
I didn't want to add more network connections (and I didn't knew the exact behavior of this, like persistent connection or something)
Have you a suggestion to the good way to create a new client ?

Reply:
This does feel like an overreach inside the SecretImportReconciler to get a private attribute.
I think this struct should have an Interface that knows how to execute a Get, so something like:
```go
type K8sReader interface {
  Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
}
```

Reply:
Got it, fixed